Expensify/App
# Update semver logic: one patch version per checklist

### Details
This changes the deploy paradigm so that we no longer bump the `PATCH` version or run a staging deploy when a checklist is locked. Instead, we bump the `PATCH` version when the checklist is first created.

### Fixed Issues
$ https://github.com/Expensify/App/issues/7166

### Tests
I tested commands locally and added automated unit tests where possible.


### QA Steps
1. Close the deploy checklist.
1. The app should have it's `PATCH` version bumped, and a staging deploy should occur./
1. The new checklist should be created with the new `PATCH` version.
1. Lock the checklist when there is no staging deploy running.
1. The `awaitStagingDeploys` action should complete quickly, and `OSBotify` should comment on the deploy checklist:

    :rocket: All staging deploys are complete, @Expensify/applauseleads please begin QA on version https://github.com/Expensify/App/releases/tag/new-build-version :rocket:
    
1. Unlock the checklist and merge another PR.
1. While the staging deploy for the other PR is running, lock the checklist again.
1. The `awaitStagingDeploys` action should poll the GH API every 90 seconds until the staging deploy completes.
1. Once the staging deploy completes, the action should complete and `OSBotify` should comment on the deploy checklist with the name comment as above, using the new build version.
